parsec/class/parsec_hash_table.c

```diff
@@ -252,15 +252,23 @@ static void parsec_hash_table_resize(parsec_hash_table_t *ht)
 {
     parsec_atomic_lock_t unlocked = PARSEC_ATOMIC_UNLOCKED;
     parsec_hash_table_head_t *head;
-    int nb_bits = ht->rw_hash->nb_bits + 1;
+    parsec_hash_table_head_t *old_head = ht->rw_hash;
+    int nb_bits = old_head->nb_bits + 1;
     assert(nb_bits < 32);
 
+    /* count the number of used buckets */
```

Comment:
There seems to be an implicit assumption that `used_buckets` is 0. If that's really the case add an assert.

Reply:
Good point, I changed the code to count into a local variable and set the `used_bucket` member once. It isn't changed we count the used buckets here and then it's only decremented afterwards.